docs/contrib/quickstart.rst

```diff
@@ -0,0 +1,123 @@
+.. _quickstart:
+
+Quick Start Guide
+=================
+
+Requirements
+------------
+
+- Java 8 or newer
+- sbt 1.5.8 or newer
+- LLVM/Clang 6.0 or newer
+
+Project Structure Overview
+--------------------------
+
+See :ref:`build`
+
+Project suffix
+--------------
+
+Most projects in ScalaNative cross-build against Scala ``2.12``, ``2.13`` and ``3``, and these projects have a suffix like ``2_12``, ``2_13`` or ``3`` to differentiate the Scala version.
+For example, ``sandbox`` has ``sandbox2_12``, ``sandbox2_13`` and ``sandbox3``.
+
+In the following we will use suffix ``3``, but remember that you can build and test for different versions using different suffixes.
+
+Commonn sbt commands
+--------------------
+
+**Common Test Commands**
+
+- ``tests3/test`` - run the unit tests for libraries on native build
+- ``tests3/testOnly org.scalanative.testsuite.javalib.util.RandomTest`` - run only the test of interest
+- ``tests3/testOnly *.RandomTest`` - run only the test of interest using wildcard
+- ``testsExt3/test`` - run the unit tests on native build, this module contains tests that requires dummy javalib implementation defined in ``javalibExtDummies``.
+- ``nir3/test`` - run the unit tests for NIR
+- ``tools3/test`` - run the unit tests of the tools: ScalaNative backend
+- ``sbtScalaNative/scripted`` - run all integration tests of the sbt plugin (this takes a while).
+- ``sbtScalaNative/scripted <test directory to run>`` - run specific integration tests of the sbt plugin. e.g. ``sbtScalaNative/scripted run/backtrace``
+    - Scripted tests are used when you need to interact with the file system, networking, or the build system that cannot be done with a unit test.
+
+**Other Test Commands**
+
+- ``testsJVM3/test`` - run ``tests3/test`` on JVM
+- ``testsExtJVM3/test`` - run ``testsExt3/test`` on JVM
+- ``test-all`` - run all tests, ideally after ``reload`` and ``clean``
```

Comment:
One thing I have noticed is that reloading will reset the version of Scala you might have configured when you started sbt.
```
% sbt "++3.3.0; shell"
...
sbt:scala-native> scalaVersion
[info] 3.3.0
sbt:scala-native> clean; reload
...
sbt:scala-native> scalaVersion
[info] 2.12.18
```
I might be doing something wrong but that definitely was a surprise for me. Maybe it deserves a mention in this guide.

Reply:
I think it's more like sbt usage? 

- You don't need to decide the Scala version to use when you launch sbt shell
  - You can just run `$ sbt` to launch sbt shell, no need for `++x.y.z` and `shell`
- You can change the scala version in sbt shell whenever you want
 
```
$ sbt
...
sbt:scala-native> scalaVersion
[info] 2.12.18
sbt:scala-native> ++3.3.0
...
sbt:scala-native> scalaVersion
[info] 3.3.0
```

I'm not sure we should add explanation for that here.
And do we want to document `test-all`, since it takes so long and I personally never used in my local development.

Reply:
> I think it's more like sbt usage?

OK, I understand.

I think I would still add something like "remember that `reload` will cause the session to be restarted fresh, meaning that your configurations (e.g., `++3.3.0`) will be reset as well."

And thanks for the explanations, by the way ;)

> And do we want to document test-all, since it takes so long and I personally never used in my local development.

I think we should. FWIW, the first thing I always try to do when I get started with an open source project is to try running the most comprehensive test suite possible.

Reply:
I think it is better to point to something a little less comprehensive to start like `tests3/test` or the tools test. The `testsJVM` will run tests against the VM so the is good to mention for `javalib`. You might want to mention the 2_13 and 2_12 versions as well. We can't assume we have sbt experts.

Reply:
I hear you but my two cents is that I still believe that there should be a way for someone cloning SN for the first time to make sure everything works as expected, even if `test-all` is a command that's almost never run in the "standard development workflow".

Reply:
You are correct but this is very much a WIP and we may have failures especially with new versions of software or different platforms - what we are trying to do is hard and 0.5.0 has not really settled down 100%.

Reply:
OK, I understand.

If we can agree that the goal is to eventually be able to run `test-all` on supported platforms, why not keep a brief record of this discussion in the docs for the next me? I would suggest something like this:

```
Once you've installed all dependencies, run the `test-all` command in `sbt` to confirm that your environment is properly configured. It will run will run every test defined in this project, which might take a while.

> Note: it is possible that some tests ran by `test-all` might not pass or compile on your environment. We are currently working on stabilizing this test suite for our supported platforms. As of this writing (<insert date>), we expect all tests to pass on Linux with LLVM X1.Y2, `sbt` X2.Y2, <insert other known dependencies>.
>
> Should `test-all` fail on your machine, you may try to run smaller test suites one after the other, specifically `test-runtime`, `test-tools`, <insert other test commands>.
> Known issues are <insert known issues>.
```

Reply:
I just added a disclaimer that `test-all` may fail https://github.com/scala-native/scala-native/pull/3496/commits/1e57feaf91b17269997c25ae6ca0aad989c5ea0a
I hope someone who know more about test-all can update the sentence later.